packages/cli/src/commands/package-dynamic-plugins/command.ts

```diff
@@ -0,0 +1,229 @@
+import { PackageRoles } from '@backstage/cli-node';
+
+import { OptionValues } from 'commander';
+import fs from 'fs-extra';
+
+import os from 'os';
+import path from 'path';
+
+import { paths } from '../../lib/paths';
+import { Task } from '../../lib/tasks';
+
+export async function command(opts: OptionValues): Promise<void> {
+  const { forceExport, preserveTempDir, tag, containerTool } = opts;
+  const workspacePackage = await fs.readJson(
+    paths.resolveTarget('package.json'),
+  );
+  const workspacePackageRole =
+    PackageRoles.detectRoleFromPackage(workspacePackage);
+  const workspacePackageRoleInfo =
+    workspacePackageRole !== undefined
+      ? PackageRoles.getRoleInfo(workspacePackageRole)
+      : undefined;
+  const isMonoRepo =
+    typeof workspacePackage.workspaces?.packages !== 'undefined';
+  // Find all plugin packages in the workspace
+  const packages = isMonoRepo
+    ? await discoverPluginPackages()
+    : [
+        {
+          packageDirectory: '.',
+          packageFilePath: 'package.json',
+          packageJson: workspacePackage,
+          packageRole: workspacePackageRole,
+          packageRoleInfo: workspacePackageRoleInfo,
+        },
+      ];
+  // run export-dynamic on each plugin package
+  for (const pluginPkg of packages) {
+    const { packageDirectory, packageFilePath } = pluginPkg;
+    if (
+      !fs.existsSync(path.join(packageDirectory, 'dist-dynamic')) ||
+      forceExport
+    ) {
+      Task.log(
+        `Running export-dynamic-plugin on plugin package ${packageFilePath}`,
+      );
+      try {
+        await Task.forCommand(`yarn export-dynamic`, { cwd: packageDirectory });
```

Comment:
Shouldn't we have different options for frontend/backend plugins?( `--no-in-place` for frontend or `--embed-as-dependencies` for backend)

Reply:
I see now, we call the user defined command, hence user must be responsible to add additional flags